lib/src/rules/unnecessary_parenthesis.dart

```diff
@@ -100,11 +101,22 @@ class _Visitor extends SimpleAstVisitor<void> {
       }
     }
 
-    if (parent is ParenthesizedExpression) {
+    // Directly wrapped into parentheses already - always report.
+    if (parent is ParenthesizedExpression ||
+        parent is InterpolationExpression ||
+        (parent is ArgumentList && parent.arguments.length == 1) ||
+        (parent is IfStatement && node == parent.condition) ||
+        (parent is WhileStatement && node == parent.condition) ||
```

Comment:
This might want to also check for the condition in a do-while statement and the expression in a switch statement or switch expression.

Reply:
@oprypin: do you want to add these cases or add a todo?

Reply:
Added DoStatement (+test), SwitchStatement (+test), SwitchExpression